mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -126,15 +180,69 @@ abstract class MLWriter extends BaseReadWrite with Logging {
     this
   }
 
+  // override for Java compatibility
+  override def session(sparkSession: SparkSession): this.type = super.session(sparkSession)
+
+  // override for Java compatibility
+  override def context(sqlContext: SQLContext): this.type = super.session(sqlContext.sparkSession)
+}
+
+/**
+ * A ML Writer which delegates based on the requested format.
+ */
+class GeneralMLWriter(stage: PipelineStage) extends MLWriter with Logging {
```

Comment:
Perhaps for another PR, but maybe we could add a method here:

```scala
  def pmml(path: String): Unit = {
    this.source = "pmml"
    save(path)
  }
```

Reply:
The follow up issue to track this is https://issues.apache.org/jira/browse/SPARK-11241